docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,357 @@
+The following table lists appropriate reactions by status code (written as ``STATUS_CODE``) and error code (written in capital letters with a link to the error code documentation).
+Fields in the error metadata are written as ``field`` in lowercase letters.
+
+.. list-table:: Command deduplication error handling with known processing time bound
+   :widths: 10 50
+   :header-rows: 1
+
+   - * Error condition
+     
+     * Reaction
+
+       
+   - * ``DEADLINE_EXCEEDED``
+     
+     * Consider the submission as lost.
+       
+       Retry from obtaining a completion offset ``OFF1`` (:ref:`step 2 <dedup-bounded-step-offset>`) and possibly increase the timeout.
+
+       
+   - * Application crashed
+     
+     * Retry by restarting the submission process from "obtaining a completion offset" (:ref:`step 2 <dedup-bounded-step-offset>`).
+
+
+   - * ``ALREADY_EXISTS`` / :ref:`DUPLICATE_COMMAND <error_code_DUPLICATE_COMMAND>`
+     
+     * The change ID has already been accepted by the ledger within the reported deduplication period.
+       The optional field ``completion_offset`` contains the precise offset.
+       The optional field ``existing_submission_id`` contains the submission ID of 
+       Report success for the ledger change.
+       
+       If desired, query the ``completion_offset`` via the :ref:`Command Completion Service <command-submission-service>` to find out about the earlier outcome.
+
+       
+   - * ``FAILED_PRECONDITION`` / :ref:`INVALID_DEDUPLICATION_PERIOD <error_code_INVALID_DEDUPLICATION_PERIOD>`
+     
+     * The specified deduplication period is longer than what the Daml ledger supports.
+       ``earliest_offset`` contains the earliest deduplication offset or ``longest_duration`` contains the longest deduplication duration that can be used (at least one of the two must be provided).
+
+       Options:
+
+       - Negotiate support for longer deduplication periods with the ledger operator.
+
+       - Set the deduplication offset to ``earliest_offset`` or the deduplication duration to ``longest_duration`` and retry from obtaining a completion offset (:ref:`step 2 <dedup-bounded-step-offset>`).
+	 This may lead to accepting the change twice within the originally intended deduplication period.
+
+	 
+   - * ``ABORTED`` / :ref:`SUBMISSION_ALREADY_IN_FLIGHT <error_code_SUBMISSION_ALREADY_IN_FLIGHT>`
+     
+       This error occurs only as an RPC response, not inside a completion event.
+       
+     * There is already another submission in flight, with the submission ID in ``existing_submission_id``.
+
+       - When you use the :ref:`Command Service <command-service>`, wait a bit and retry from submitting the command (:ref:`step 3 <dedup-bounded-step-submit>`).
+
+	 Since the in-flight submission might still be rejected, (repeated) resubmission ensures that you (eventually) learns the outcome:
```

Comment:
```suggestion
	 Since the in-flight submission might still be rejected, (repeated) resubmission ensures that you (eventually) learn the outcome:
```

Reply:
Fixed.